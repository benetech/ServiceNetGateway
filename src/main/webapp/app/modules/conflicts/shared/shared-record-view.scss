@import '../../../app-variables';
@import '../../../bootstrap-variables';

.title {
  margin-top: 15px;
}

.blueBorder {
  border-color: $blue-card-text;
  border-width: 3px;
}

.grayBorder {
  border-color: $gray-card-accent;
  border-width: 3px;
}

.icon-conflicting {
  color: $blue-card-text;
}

.icon-excluded {
  color: $gray-card-accent;
}

.icon-text {
  margin-top: 8px;
  margin-left: -45px;
  position: relative;
}

.icon-textarea {
  margin-top: 8px;
  margin-left: -45px !important;
  position: relative;
}

.icon-checkbox {
  margin-left: 5px;
}

.tooltip-inner {
  color: $tooltip-text-secondary;
  background-color: $tooltip-bg;
  box-shadow: 2px 2px 2px $tooltip-text-secondary;
}

.tooltip {
  margin-left: -10px;
}

.tooltip.show {
  opacity: 1 !important;
}

.half-rule {
  margin-top: 8px;
  margin-bottom: 8px;
  background-color: $tooltip-text-secondary;
}

.suggested {
  color: $tooltip-text-primary;
  text-align: left;
}

.secondary {
  color: $tooltip-text-secondary;
  margin: 0;
}

.jumbotron {
  margin-top: 15px;
  padding: 0;
  background-color: transparent;
  border: 0;
}

.jumbotron-header {
  padding: 10px;
  margin: 0;
  background-color: $compare-header-bg;
  border: 1px $compare-border solid;
  border-bottom: 0;
}

.jumbotron-content {
  padding: 10px;
  margin: 0;
  background-color: $compare-content-bg;
  border: 1px $compare-border solid;
  max-height: 460px;
  overflow-y: scroll;
}

.jumbotron-header-title {
  color: rgb(109, 108, 108); //secondary
}

.jumbotron-header-text {
  margin: 0;
}

.review-title {
  margin-bottom: 0;
}

.collapseBtn {
  display: inline;
  cursor: pointer;
}

.expandBtn {
  margin-top: 15px;
  cursor: pointer;
}

.collapseIcon {
  display: inline;
  margin-right: 5px;
}

.chainIcon {
  display: inline;
  margin-left: 5px;
}

.matchToggle {
  float: right;
}

.changeRecordSelect {
  padding-left: 0px;
}

.copy-to-clipboard-icon {
  position: absolute;
  right: 32px;
  margin-top: 6px;
  cursor: pointer;
  visibility: hidden;
  opacity: 0;
  transition: visibility 0s, opacity 0.5s linear;
  color: $gray-card-accent;
  padding: 0 5px;
}

.url-link {
  position: absolute;
  left: 25px;
  margin-top: 6px;
  cursor: pointer;
  visibility: visible;
  color: $blue-card-text;
}

.url-icon {
  margin-right: 10px;
}

.input-field-container {
  &:hover {
    .copy-to-clipboard-icon {
      visibility: visible;
      opacity: 1;
    }
  }
}

.tooltip-clip-inner {
  color: white !important;
  background-color: $tooltip-dark-bg !important;
}

.tooltip-clip {
  top: 5px !important;
  .arrow::before {
    border-bottom-color: $tooltip-dark-bg !important;
  }
}

span[role='button'] {
  padding: 5px;
  cursor: pointer;
}

.text-blue {
  color: #158cba;
}

.back-to-top {
  div:first-child {
    margin-right: 10px;
    text-align: center;
    height: 100px;
    width: 100px;
    border-radius: 100px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    &:hover {
      background-color: $transparent005;
    }
  }
}

.dismiss-modal {
  background-color: $modal-dark;
  color: $white;

  .close {
    color: $white;
  }

  .modal-header {
    border: none;
  }

  .modal-footer {
    border: none;
  }
}

.close-button,
.close-button:hover,
.close-button:active,
.close-button:focus {
  background-color: transparent;
  color: $white;
  border: none;
  box-shadow: none;
}

.taxonomy-pills {
  font-size: 1.25em;
  span {
    margin-right: 0.5em;
  }
}
