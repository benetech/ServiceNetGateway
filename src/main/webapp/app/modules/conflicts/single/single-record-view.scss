@import '../../../app-variables';
@import '../../../bootstrap-variables';

.title {
  margin-top: 15px;
}

.blueBorder {
  border-color: $blue-card-text;
  border-width: 3px;
}

.grayBorder {
  border-color: $gray-card-accent;
  border-width: 3px;
}

.icon-conflicting {
  color: $blue-card-text;
}

.icon-excluded {
  color: $gray-card-accent;
}

.icon-text {
  margin-top: 8px;
  margin-left: -45px;
  position: relative;
}

.icon-textarea {
  margin-top: 8px;
  margin-left: -55px;
  position: relative;
}

.icon-checkbox {
  margin-left: 5px;
}

.tooltip-inner {
  color: $tooltip-text-secondary;
  background-color: $tooltip-bg;
  box-shadow: 2px 2px 2px $tooltip-text-secondary;
}

.tooltip {
  margin-left: 10px;
}

.half-rule {
  margin-top: 8px;
  margin-bottom: 8px;
  background-color: $tooltip-text-secondary;
}

.suggested {
  color: $tooltip-text-primary;
  text-align: left;
}

.secondary {
  color: $tooltip-text-secondary;
  margin: 0;
}

.jumbotron {
  margin-top: 15px;
  padding: 0;
  background-color: transparent;
  border: 0;
}

.jumbotron-header {
  padding: 10px;
  margin: 0;
  background-color: $compare-header-bg;
  border: 1px $compare-border solid;
  border-bottom: 0;
}

.jumbotron-content {
  padding: 0 !important;
  margin: 0;
  background-color: $compare-content-bg;
  border: 1px $compare-border solid;
}

.jumbotron-header-title {
  color: rgb(102, 101, 101); //secondary
}

.jumbotron-header-text {
  margin: 0;
}

.review-title {
  margin-bottom: 0;
}

.collapseBtn {
  display: inline;
  cursor: pointer;
}

.collapseIcon {
  display: inline;
  margin-right: 5px;
}

.match-details {
  display: flex;
  flex-direction: row;
}

.match-details > * {
  flex: 1;
}

.jumbotron-content > a {
  color: black;
  text-decoration: none;
}

.match > h5 {
  color: #6d6c6c;
}

.match {
  display: block;
  padding: 18px 10px 18px 2rem;
}

.match:not(:last-child) {
  border-bottom: 1px black solid;
}

.no-match-text {
  display: flex;
  height: 100%;
  justify-content: center;
  align-items: center;
}
