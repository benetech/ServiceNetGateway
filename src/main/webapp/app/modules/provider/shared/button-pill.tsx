import React from 'react';
import { Translate } from 'react-jhipster';

const ButtonPill = props => (
  <div className="button-pill" onClick={props.onClick}>
    <div className={`d-inline ${props.additionalClass}`}>
      <b>
        <Translate contentKey={props.translate} />
      </b>
    </div>
  </div>
);

export default ButtonPill;
