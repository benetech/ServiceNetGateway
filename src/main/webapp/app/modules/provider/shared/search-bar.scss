@import '../app-variables';
@import '../filter-card';

.search-from {
  background-color: white;
}

.search-from {
  height: 30px;
  display: flex;
  padding: 2px 4px;
  box-shadow: 1px 1px 2px 1px rgba(0, 0, 0, 0.3);
  align-items: center;
  border-radius: 25px;
}

.search-input-container {
  flex: 1;
  margin-left: 8px;
}

input[type='search']::-webkit-search-cancel-button {
  -webkit-appearance: none;
}

.search-input {
  outline: none;
  font: inherit;
  color: currentColor;
  width: 100%;
  border: 0;
  height: 1.1876em;
  margin: 0;
  display: block;
  padding: 6px 0 7px;
  min-width: 0;
  background: none;
  box-sizing: content-box;
  animation-name: mui-auto-fill-cancel;
  -webkit-tap-highlight-color: transparent;
}

.search-button {
  flex: 0 0 auto;
  color: rgba(0, 0, 0, 0.54);
  padding: 8px;
  overflow: visible;
  font-size: 1.5rem;
  text-align: center;
  transition: background-color 150ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 50%;
  background: rgba(255, 255, 255, 0);
  border: none;
}

.search-icon {
  width: 100%;
  display: flex;
  align-items: inherit;
  justify-content: inherit;
}

@media only screen and (max-width: 768px) {
  .search-bar {
    padding-right: 0;
    margin-bottom: 0.5rem;
    margin-top: 0.25rem;

    .search-from {
      width: 100%;
    }
  }
}
