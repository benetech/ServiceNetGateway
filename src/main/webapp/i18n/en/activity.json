{
  "serviceNetApp": {
    "activity": {
      "home": {
        "organizations": "Organizations",
        "services": "Services",
        "title": "Activities",
        "createLabel": "Create a new Activity",
        "createOrEditLabel": "Create or edit a Activity",
        "leftColumnTitle": "Record Name",
        "rightColumnTitle": "Matches found",
        "filter": {
          "toggle": "Filter",
          "savedFilters": "Saved Filters",
          "saveFilter": "Save Filter",
          "deleteFilter": "Delete Filter",
          "deleteQuestion": "Are you sure you want to delete this Activity Filter?",
          "deleteSuccess": "Filter was deleted successfully",
          "deleteError": "An error occurred while deleting filter",
          "filterName": "Filter Name",
          "saveSuccess": "Filter was saved successfully",
          "saveError": "Error occurred while saving filter",
          "city": "City",
          "county": "County",
          "zip": "ZIP",
          "taxonomy": "Taxonomy",
          "partner": "Partner Sources",
          "applyFilter": "Apply",
          "resetFilter": "Reset",
          "searchFor": "Search For",
          "organization": "Organizations",
          "services": "Services",
          "locations": "Locations",
          "dateFilter": "Update date",
          "searchFields": "Search fields",
          "from": "From",
          "to": "To",
          "date": {
            "last7Days": "Last 7 Days",
            "last30Days": "Last 30 Days",
            "dateRange": "Date Range"
          },
          "onlyShowMatching": "Only show matching",
          "onlyShowHighlyMatched": "Only show highly-matched Organizations",
          "applyLocationSearch": "Search near selected location",
          "fields": {
            "name": "Name",
            "alternateName": "Alternate name",
            "description": "Description",
            "email": "E-mail",
            "url": "Website",
            "taxStatus": "Tax status",
            "taxId": "Tax ID",
            "legalStatus": "Legal status",
            "externalDbId": "External database ID",
            "status": "Status",
            "interpretationServices": "Interpretation Services",
            "applicationProcess": "Application Process",
            "waitTime": "Wait time",
            "fees": "Fees",
            "accreditations": "Accreditations",
            "licenses": "Licenses",
            "type": "Type",
            "providerName": "Provider Name",
            "transportation": "Transportation",
            "phone": "Phone",
            "contactName": "Contact Name",
            "contactPhone": "Contact Phone",
            "eligibility": "Eligibility",
            "requiredDocument": "Required Document",
            "language": "Language",
            "physicalAddress": "Physical Address",
            "postalAddress": "Postal Address",
            "accessibility": "Accessibility"
          },
          "error": {
            "specifyDates": "Please specify a From and an Until date.",
            "specifyFromDate": "Please specify a From date.",
            "specifyToDate": "Please specify an Until date.",
            "untilDateEarlierThanFromDate": "Please specify a From date that is on or before the Until date.",
            "invalidDates": "Please specify valid From and Until dates."
          },
          "radius": "Search in radius",
          "locationChooser": "Location chooser",
          "filterOptions": "Filter options",
          "setMyCurrentLocation": "Set my current location"
        },
        "sort": {
          "toggle": "Sort",
          "similarity": "Match Similarity",
          "recommended": "Recommended",
          "recent": "Recently Updated"
        },
        "search": {
          "placeholder-ORGANIZATION": "Search for organizations, leave blank to search all records",
          "placeholder-SERVICES": "Search for services, leave blank to search all records",
          "placeholder-LOCATIONS": "Search for locations, leave blank to search all records"
        }
      },
      "created": "A new Activity is created with identifier {{ param }}",
      "updated": "A Activity is updated with identifier {{ param }}",
      "deleted": "A Activity is deleted with identifier {{ param }}",
      "delete": {
        "question": "Are you sure you want to delete Activity {{ id }}?"
      },
      "detail": {
        "title": "Activity"
      },
      "organization": "Organization",
      "noConflicts": "0 Partner conflicts with this record",
      "conflict": "Conflict",
      "empty": "There are no recent activities",
      "unresolved": {
        "title": "Recent Service Net Activity",
        "conflictPlusZero": " has a conflicting {{ fieldName }}"
      },
      "lastPartnerUpdate": "Last Partner Update ",
      "hideRecord": "Done with this record",
      "serviceProviderOrgTooltip": "This record is maintained by the service provider"
    }
  }
}
