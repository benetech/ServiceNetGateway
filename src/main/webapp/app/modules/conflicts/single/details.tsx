import React from 'react';
import './single-record-view.scss';
import { connect } from 'react-redux';
import { RouteComponentProps, Link } from 'react-router-dom';
import { IActivityRecord } from 'app/shared/model/activity-record.model';
import { IOrganizationMatch } from 'app/shared/model//organization-match.model';
import { OrganizationDetails } from '../shared/components/organization-details';
import LocationsDetails from '../shared/components/location/locations-details';
import ServicesDetails from '../shared/components/service/services-details';
import { ContactsDetails } from '../shared/components/contact/contacts-details';
import { Col, Jumbotron } from 'reactstrap';
import { translate, Translate } from 'react-jhipster';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import ReactGA from 'react-ga';
import _ from 'lodash';
import { getNotHiddenMatchesByOrg, getPartnerRecords } from 'app/modules/conflicts/shared/shared-record-view.reducer';
import { SYSTEM_ACCOUNTS } from 'app/config/constants';
import IconSpan from 'app/shared/layout/icon-span';
import OwnerInfo from 'app/shared/layout/owner-info';

export interface ISingleRecordViewProp extends StateProps, DispatchProps, RouteComponentProps<{}> {
  activity: IActivityRecord;
  organizationMatches?: IOrganizationMatch[];
  dismissedMatches?: IOrganizationMatch[];
  isBaseRecord: boolean;
  showClipboard: boolean;
  orgId: string;
}

export interface ISingleRecordViewState {
  activeTab: string;
  isAreaOpen: boolean;
}

export class Details extends React.Component<ISingleRecordViewProp, ISingleRecordViewState> {
  state: ISingleRecordViewState = {
    activeTab: '1',
    isAreaOpen: false
  };

  static getLocationMatches = matches => {
    const locationMatches = {};
    const keys = _.map(matches, match => _.get(match, 'locationMatches'));
    _.forEach(keys, a => {
      _.forEach(a, (c, d) => {
        if (_.isEmpty(locationMatches[d])) {
          locationMatches[d] = [];
        }
        locationMatches[d].push(...c);
        _.flattenDepth(locationMatches[d], 2);
      });
    });
    return locationMatches;
  };

  componentDidMount() {
    this.props.getPartnerRecords(this.props.orgId);
  }

  toggle = tab => {
    if (this.state.activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  };

  toggleAreaOpen = () => {
    this.setState({
      isAreaOpen: !this.state.isAreaOpen
    });
  };

  handleMatchClick = () => {
    ReactGA.event({ category: 'UserActions', action: 'Clicking On Side By Side View' });
  };

  render() {
    const { orgId, organizationMatches, dismissedMatches } = this.props;
    const firstMatch = organizationMatches.length !== 0 ? organizationMatches[0] : null;

    const noMatchSection = (
      <Col md="6">
        <h3 className="no-match-text">
          <Translate contentKey="singleRecordView.noMatches" />
        </h3>
      </Col>
    );

    const matchSection = firstMatch ? (
      <Col sm="3">
        <Jumbotron className="jumbotron">
          <div className="jumbotron-header">
            <h4 className="jumbotron-header-title">
              <Translate contentKey="singleRecordView.details.compare.title" />
            </h4>
            <p className="jumbotron-header-text">
              {organizationMatches.length}
              {organizationMatches.length === 1 ? (
                <Translate contentKey="singleRecordView.details.compare.singleHeader" />
              ) : (
                <Translate contentKey="singleRecordView.details.compare.multiHeader" />
              )}
            </p>
          </div>
          <div className="jumbotron-content">
            {organizationMatches
              ? organizationMatches
                  .sort((om1, om2) => Object.keys(om2.locationMatches).length - Object.keys(om1.locationMatches).length)
                  .map(match => (
                    <Link
                      key={match.id}
                      to={`/multi-record-view/${match.organizationRecordId}/${match.partnerVersionId}`}
                      onClick={this.handleMatchClick}
                      className={`match ${match.providerName === SYSTEM_ACCOUNTS.SERVICE_PROVIDER ? 'pl-0' : ''}`}
                    >
                      <IconSpan visible={match.providerName === SYSTEM_ACCOUNTS.SERVICE_PROVIDER}>
                        <h5>{match.partnerVersionName}</h5>
                        <div className="match-details">
                          <div>
                            {match.providerName === SYSTEM_ACCOUNTS.SERVICE_PROVIDER ? (
                              <OwnerInfo owner={match.owner} direction="right" />
                            ) : (
                              match.providerName
                            )}
                          </div>
                          <div className="text-right">
                            {match.numberOfLocations}{' '}
                            <Translate
                              contentKey={'singleRecordView.details.compare.' + (match.numberOfLocations > 1 ? 'locations' : 'location')}
                            />{' '}
                            {' / '}
                            {Object.keys(match.locationMatches).length}{' '}
                            <Translate
                              contentKey={
                                'singleRecordView.details.compare.' + (Object.keys(match.locationMatches).length > 1 ? 'matches' : 'match')
                              }
                            />
                          </div>
                        </div>
                        <div className="text-right" title={translate('singleRecordView.details.compare.sinceLastUpdatedTooltip')}>
                          {match.freshness} <Translate contentKey="singleRecordView.details.compare.sinceLastUpdated" />
                        </div>
                      </IconSpan>
                    </Link>
                  ))
              : null}
            <Link to={`/all-records-view/${orgId}`} onClick={this.handleMatchClick} className="match">
              <h5>Show all matches</h5>
            </Link>
          </div>
        </Jumbotron>
      </Col>
    ) : !dismissedMatches.length ? null : (
      <div>
        <Link to={`/dismissed-matches/${this.props.orgId}`}>
          <Translate contentKey="singleRecordView.details.compare.viewDismissedMatches" />
        </Link>
      </div>
    );

    const sideSection =
      organizationMatches && organizationMatches.length === 0 && dismissedMatches.length === 0 ? noMatchSection : matchSection;

    const columnSize = 6;
    const { isAreaOpen } = this.state;
    const { partnerRecords, matches } = this.props;
    const locationMatches = Details.getLocationMatches(matches);
    const recordMatch = partnerRecords.length && _.find(matches, m => m.partnerVersionId === partnerRecords[0].organization.id);
    return (
      <div>
        <OrganizationDetails {...this.props} sideSection={sideSection} columnSize={columnSize} />
        <h5 className="expandBtn">
          <div className="collapseBtn" onClick={this.toggleAreaOpen}>
            <div className="collapseIcon">
              <FontAwesomeIcon size="xs" icon={isAreaOpen ? 'angle-up' : 'angle-down'} />
            </div>
            <Translate contentKey={isAreaOpen ? 'singleRecordView.details.collapseAll' : 'singleRecordView.details.expandAll'} />
          </div>
        </h5>
        {isAreaOpen ? (
          <div>
            <LocationsDetails
              {...this.props}
              locations={this.props.activity.locations}
              columnSize={columnSize}
              locationMatches={recordMatch && locationMatches}
              isAreaOpen
            />
            <ServicesDetails
              {...this.props}
              services={this.props.activity.services}
              columnSize={columnSize}
              serviceMatches={recordMatch && recordMatch.serviceMatches}
              isAreaOpen
            />
            <ContactsDetails {...this.props} contacts={this.props.activity.contacts} columnSize={columnSize} isAreaOpen />
          </div>
        ) : null}
        {isAreaOpen ? null : (
          <div>
            <LocationsDetails
              {...this.props}
              locations={this.props.activity.locations}
              columnSize={columnSize}
              locationMatches={recordMatch && locationMatches}
              isAreaOpen={false}
            />
            <ServicesDetails
              {...this.props}
              services={this.props.activity.services}
              columnSize={columnSize}
              serviceMatches={recordMatch && recordMatch.serviceMatches}
              isAreaOpen={false}
            />
            <ContactsDetails {...this.props} contacts={this.props.activity.contacts} columnSize={columnSize} isAreaOpen={false} />
          </div>
        )}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  partnerRecords: state.sharedRecordView.partnerRecords,
  matches: state.sharedRecordView.matches
});

const mapDispatchToProps = { getPartnerRecords, getNotHiddenMatchesByOrg };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Details);
