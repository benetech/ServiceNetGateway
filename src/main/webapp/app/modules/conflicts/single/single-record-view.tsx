import 'filepond/dist/filepond.min.css';
import './single-record-view.scss';

import React from 'react';
import { connect } from 'react-redux';
import { Row, Col } from 'reactstrap';
import Tabs from './tabs';
import { getBaseRecord, getMatches } from '../shared/shared-record-view.reducer';
import { RouteComponentProps } from 'react-router-dom';
import { TextFormat, Translate } from 'react-jhipster';
import { APP_DATE_FORMAT } from 'app/config/constants';
import { getUser } from 'app/modules/administration/user-management/user-management.reducer';

export interface ISingleRecordViewProp extends StateProps, DispatchProps, RouteComponentProps<{}> {
  showClipboard: boolean;
}

export interface ISingleRecordViewState {
  match: any;
}

export class SingleRecordView extends React.Component<ISingleRecordViewProp, ISingleRecordViewState> {
  state: ISingleRecordViewState = {
    match: null
  };

  componentDidMount() {
    this.props.getBaseRecord(this.props.orgId);
    this.props.getMatches(this.props.orgId);
    this.props.getUser(this.props.account.login);
  }

  render() {
    const { activityRecord, user } = this.props;
    const content = activityRecord ? (
      <Row>
        <Col>
          <h2>{activityRecord.organization.name}</h2>
          <h5>
            {activityRecord.organization.accountId === user.systemAccountId ? (
              <Translate contentKey="multiRecordView.yourData" />
            ) : (
              <span>
                <Translate contentKey="singleRecordView.partnerName" /> {activityRecord.organization.accountName}
              </span>
            )}
          </h5>
          <h5>
            <Translate contentKey="multiRecordView.lastCompleteReview" />
            {activityRecord.organization.lastVerifiedOn ? (
              <TextFormat value={activityRecord.organization.lastVerifiedOn} type="date" format={APP_DATE_FORMAT} blankOnInvalid />
            ) : (
              <Translate contentKey="multiRecordView.unknown" />
            )}
          </h5>
          <h5>
            <Translate contentKey="multiRecordView.lastUpdated" />
            {activityRecord.lastUpdated ? (
              <TextFormat value={activityRecord.lastUpdated} type="date" format={APP_DATE_FORMAT} blankOnInvalid />
            ) : (
              <Translate contentKey="multiRecordView.unknown" />
            )}
          </h5>
          <Tabs activity={activityRecord} {...this.props} />
        </Col>
      </Row>
    ) : (
      <Row>
        <Col>
          <h2>Loading...</h2>
        </Col>
      </Row>
    );

    return content;
  }
}

const mapStateToProps = (storeState, { match }: ISingleRecordViewState) => ({
  orgId: match.params.orgId,
  activityRecord: storeState.sharedRecordView.baseRecord,
  organizationMatches: storeState.sharedRecordView.matches,
  dismissedMatches: storeState.sharedRecordView.dismissedMatches,
  account: storeState.authentication.account,
  user: storeState.userManagement.user
});

const mapDispatchToProps = { getBaseRecord, getMatches, getUser };

type StateProps = ReturnType<typeof mapStateToProps>;
type DispatchProps = typeof mapDispatchToProps;

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(SingleRecordView);
