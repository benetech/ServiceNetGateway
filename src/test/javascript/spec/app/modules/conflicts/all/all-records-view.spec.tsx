import React from 'react';
import AllRecordsView from 'app/modules/conflicts/all/all-records-view';
import { Details as DetailClass } from 'app/modules/conflicts/single/details';
import { commonProps } from '../shared/data';
import activity from '../shared/data/activity.json';
import { wrapper } from '../shared/wrapper';

describe('Multiple record view', () => {
  let mountedWrapper;

  const props = {
    ...commonProps,
    match: { params: { orgId: activity.organization.id } },
    matchNumber: 0,
    showDismissModal: false,
    showSuccessModal: false,
    dismissError: false,
    locationMatches: DetailClass.getLocationMatches(commonProps.matches),
    selectedLocation: null,
    locationsHaveMatch: true,
    matchingLocation: null,
    selectedMatch: null,
    fieldSettingsExpanded: false,
    tooltipOpen: false,
    loadingPartner: false
  };

  it('renders correctly', () => {
    mountedWrapper = wrapper(mountedWrapper, <AllRecordsView {...props} />);
    // the created snapshot must be committed to source control
    expect(mountedWrapper).toMatchSnapshot();
  });
});
