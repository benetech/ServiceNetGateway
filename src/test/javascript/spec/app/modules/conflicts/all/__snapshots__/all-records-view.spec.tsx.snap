// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple record view renders correctly 1`] = `
<div
  className="all-records-view shared-record-view"
>
  <div
    className="fields-display-settings-btn"
    id="fields-display-settings-btn"
    onClick={[Function]}
  >
    <span>
      <span />
       
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-cogs fa-w-20 fa-lg "
        data-icon="cogs"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 640 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M512.1 191l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0L552 6.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zm-10.5-58.8c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.7-82.4 14.3-52.8 52.8zM386.3 286.1l33.7 16.8c10.1 5.8 14.5 18.1 10.5 29.1-8.9 24.2-26.4 46.4-42.6 65.8-7.4 8.9-20.2 11.1-30.3 5.3l-29.1-16.8c-16 13.7-34.6 24.6-54.9 31.7v33.6c0 11.6-8.3 21.6-19.7 23.6-24.6 4.2-50.4 4.4-75.9 0-11.5-2-20-11.9-20-23.6V418c-20.3-7.2-38.9-18-54.9-31.7L74 403c-10 5.8-22.9 3.6-30.3-5.3-16.2-19.4-33.3-41.6-42.2-65.7-4-10.9.4-23.2 10.5-29.1l33.3-16.8c-3.9-20.9-3.9-42.4 0-63.4L12 205.8c-10.1-5.8-14.6-18.1-10.5-29 8.9-24.2 26-46.4 42.2-65.8 7.4-8.9 20.2-11.1 30.3-5.3l29.1 16.8c16-13.7 34.6-24.6 54.9-31.7V57.1c0-11.5 8.2-21.5 19.6-23.5 24.6-4.2 50.5-4.4 76-.1 11.5 2 20 11.9 20 23.6v33.6c20.3 7.2 38.9 18 54.9 31.7l29.1-16.8c10-5.8 22.9-3.6 30.3 5.3 16.2 19.4 33.2 41.6 42.1 65.8 4 10.9.1 23.2-10 29.1l-33.7 16.8c3.9 21 3.9 42.5 0 63.5zm-117.6 21.1c59.2-77-28.7-164.9-105.7-105.7-59.2 77 28.7 164.9 105.7 105.7zm243.4 182.7l-8.2 14.3c-3 5.3-9.4 7.5-15.1 5.4-11.8-4.4-22.6-10.7-32.1-18.6-4.6-3.8-5.8-10.5-2.8-15.7l8.2-14.3c-6.9-8-12.3-17.3-15.9-27.4h-16.5c-6 0-11.2-4.3-12.2-10.3-2-12-2.1-24.6 0-37.1 1-6 6.2-10.4 12.2-10.4h16.5c3.6-10.1 9-19.4 15.9-27.4l-8.2-14.3c-3-5.2-1.9-11.9 2.8-15.7 9.5-7.9 20.4-14.2 32.1-18.6 5.7-2.1 12.1.1 15.1 5.4l8.2 14.3c10.5-1.9 21.2-1.9 31.7 0l8.2-14.3c3-5.3 9.4-7.5 15.1-5.4 11.8 4.4 22.6 10.7 32.1 18.6 4.6 3.8 5.8 10.5 2.8 15.7l-8.2 14.3c6.9 8 12.3 17.3 15.9 27.4h16.5c6 0 11.2 4.3 12.2 10.3 2 12 2.1 24.6 0 37.1-1 6-6.2 10.4-12.2 10.4h-16.5c-3.6 10.1-9 19.4-15.9 27.4l8.2 14.3c3 5.2 1.9 11.9-2.8 15.7-9.5 7.9-20.4 14.2-32.1 18.6-5.7 2.1-12.1-.1-15.1-5.4l-8.2-14.3c-10.4 1.9-21.2 1.9-31.7 0zM501.6 431c38.5 29.6 82.4-14.3 52.8-52.8-38.5-29.6-82.4 14.3-52.8 52.8z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </span>
  </div>
  <label
    className="sr-only"
    htmlFor="settings"
  >
    <span />
  </label>
  <div
    className="fields-display-settings-selector css-2b097c-container"
    onKeyDown={[Function]}
  >
    <div
      className=" css-yk16xz-control"
      onMouseDown={[Function]}
      onTouchEnd={[Function]}
    >
      <div
        className=" css-1hwfws3"
      >
        <div
          className=" css-1wa3eu0-placeholder"
        >
          Select...
        </div>
        <div
          className="css-b8ldur-Input"
        >
          <div
            className=""
            style={
              Object {
                "display": "inline-block",
              }
            }
          >
            <input
              aria-autocomplete="list"
              autoCapitalize="none"
              autoComplete="off"
              autoCorrect="off"
              disabled={false}
              id="settings"
              onBlur={[Function]}
              onChange={[Function]}
              onFocus={[Function]}
              spellCheck="false"
              style={
                Object {
                  "background": 0,
                  "border": 0,
                  "boxSizing": "content-box",
                  "color": "inherit",
                  "fontSize": "inherit",
                  "label": "input",
                  "opacity": 1,
                  "outline": 0,
                  "padding": 0,
                  "width": "1px",
                }
              }
              tabIndex="0"
              type="text"
              value=""
            />
            <div
              style={
                Object {
                  "height": 0,
                  "left": 0,
                  "overflow": "scroll",
                  "position": "absolute",
                  "top": 0,
                  "visibility": "hidden",
                  "whiteSpace": "pre",
                }
              }
            >
              
            </div>
          </div>
        </div>
      </div>
      <div
        className=" css-1hb7zxy-IndicatorsContainer"
      >
        <span
          className=" css-1okebmr-indicatorSeparator"
        />
        <div
          aria-hidden="true"
          className=" css-tlfecz-indicatorContainer"
          onMouseDown={[Function]}
          onTouchEnd={[Function]}
        >
          <svg
            aria-hidden="true"
            className="css-6q0nyr-Svg"
            focusable="false"
            height={20}
            viewBox="0 0 20 20"
            width={20}
          >
            <path
              d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
  <div
    className="row flex-row flex-nowrap row"
  >
    <div
      className="col"
    >
      <h2>
        Loading...
      </h2>
    </div>
    <div
      className="partner-records row flex-row flex-nowrap  col"
    >
      <div
        className="col"
      >
        <h2>
          Loading...
        </h2>
      </div>
    </div>
  </div>
</div>
`;
